utils/src/ooinstall/cli_installer.py

```diff
@@ -742,6 +744,15 @@ def get_hosts_to_run_on(oo_cfg, callback_facts, unattended, force, verbose):
 
     return hosts_to_run_on, callback_facts
 
+def set_infra_nodes(hosts):
+    if all(host.is_master() for host in hosts):
+        infra_list = hosts
+    else:
+        infra_list = hosts[:2]
```

Comment:
You're right. i refactored my own code and left out the nodes-only part. Fixing.